src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const result = [ ];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties' :
+        Object.assign(stateCopy, action.extraData);
+        result.push({ ...stateCopy });
```

Comment:
You repeat this push action for each case. Just move it after the switch/case construction and call it only once